### ISSUE TITLE
The Referer header is not preserved when a GM_xmlhttpRequest request is redirected

### ISSUE THREAD

**Issue by techmovie** [Tampermonkey/tampermonkey#2179]
## Expected Behavior

Referer header is preserved when a GM_xmlhttpRequest request is redirected

## Actual Behavior

The Referer header is not preserved when a GM_xmlhttpRequest request is redirected.  

## Specifications

- Chromium: (128.0.6613.139)
- TM: 5.3.0 (When I switch the TM version to 5.1.1, the Referer header can be preserved)
- OS: MacOS 15.0

## Script

```js
// ==UserScript==
// @name referer header issue
// @version 1.0
// @description try to take over the world!
// @author Me
// @include *
// @grant GM_xmlhttpRequest
// ==/UserScript==

const fetch = (url, options) => {
  return new Promise((resolve, reject) => {
    GM_xmlhttpRequest({
      method: 'GET',
      url,
      responseType: 'json',
      ...options,
      onload: (res) => {
        const { statusText, status, response } = res;
        if (status !== 200) {
          reject(new Error(statusText || `${status}`));
        } else {
          resolve(response);
        }
      },
      ontimeout: () => {
        reject(new Error('timeout'));
      },
      onerror: (error) => {
        reject(error);
      },

    });
  });
};

const url = 'https://m.douban.com/rexxar/api/v2/movie/36205404/?for_mobile=1';
const options = {
    headers: {
        Referer: 'https://m.douban.com/movie/subject/36205404',
    },
};

fetch(url, options).then((res) => {
    console.log(res);
}).catch((error) => {
    console.error(error);
});



```

**Comment by derjanb** [comment]
At the moment I think that this is a [Chrome bug](https://github.com/w3c/webextensions/issues/694). Will add a workaround though.

**Comment by derjanb** [comment]
Should be fixed at 5.3.6212 ([crx](https://tampermonkey.net/crx/tampermonkey_beta_current.crx)|xpi in review)

Please download the crx file linked above and drag and drop it to the extensions page `chrome://extensions` (after you've enabled 'Developer Mode').

For a quick fix please export your settings and scripts as zip or (JSON) file at the "Utilities" tab and import it back at the fixed BETA version.
